### PR TITLE
EVG-15341: Return 404 when error encountered finding merged project ref

### PR DESCRIPTION
**Pull request by hadjri**
[EVG-15341](https://jira.mongodb.org/browse/EVG-15341)

### Description 
Currently all routes that use urlVarsToProjectScopes in the middleware such as this [endpoint](https://github.com/evergreen-ci/evergreen/wiki/REST-V2-Usage#get-versions-for-a-project) return a 500 internal server error when a merged project ref is unable to be found. Change has been made to return a 404 in this case instead.
### Testing 
Deployed to staging and confirmed that the route now returns 404 for an invalid project.  Attached screenshot.

### CODE REVIEW

**Comment by hadjri** [comment]
Screen Shot 2021-09-21 at 12 51 33 PM

File: rest/route/middleware.go [hunk 1]
```diff
@@ -631,7 +631,7 @@ func urlVarsToProjectScopes(r *http.Request) ([]string, int, error) {
 
 	projectRef, err := model.FindMergedProjectRef(projectID)
 	if err != nil {
-		return nil, http.StatusInternalServerError, errors.WithStack(err)
+		return nil, http.StatusNotFound, errors.WithStack(err)
```
Comment (ablack12) [inline rest/route/middleware.go:634]: I actually have PR up (although I need to fix a test) to change FindMergedProjectRef to return `nil, nil` if not found, instead of an error, which is more consistent. In that case, this _should_ be http.StatusInternalServerError, but since that PR is going to take some time I can just change this back in that PR once this is merged.

**Review by ablack12 — APPROVED** [review]
